src/OmniSharp.Cake/Services/RequestHandlers/Navigation/GotoDefinitionHandler.cs

```diff
@@ -0,0 +1,135 @@
+using System;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.Text;
+using OmniSharp.Cake.Extensions;
+using OmniSharp.Mef;
+using OmniSharp.Models.GotoDefinition;
+using OmniSharp.Models.Metadata;
+using OmniSharp.Roslyn;
+using OmniSharp.Utilities;
+
+namespace OmniSharp.Cake.Services.RequestHandlers.Navigation
+{
+    [OmniSharpHandler(OmniSharpEndpoints.GotoDefinition, Constants.LanguageNames.Cake), Shared]
+    public class GotoDefinitionHandler : CakeRequestHandler<GotoDefinitionRequest, GotoDefinitionResponse>
+    {
+        private readonly MetadataHelper _metadataHelper;
+
+        [ImportingConstructor]
+        public GotoDefinitionHandler(
+            OmniSharpWorkspace workspace,
+            MetadataHelper metadataHelper)
+            : base(workspace)
+        {
+            _metadataHelper = metadataHelper ?? throw new ArgumentNullException(nameof(metadataHelper));
+        }
+
+        protected override async Task<GotoDefinitionResponse> TranslateResponse(GotoDefinitionResponse response, GotoDefinitionRequest request)
+        {
+            if (string.IsNullOrEmpty(response.FileName) ||
+                !response.FileName.Equals(Constants.Paths.Generated))
+            {
+                if (PlatformHelper.IsWindows && !string.IsNullOrEmpty(response.FileName))
+                {
+                    response.FileName = response.FileName.Replace('/', '\\');
+                }
+                return response;
+            }
+
+            if (!request.WantMetadata)
+            {
+                return new GotoDefinitionResponse();
+            }
+
+            return await GetAliasFromMetadataAsync(new GotoDefinitionRequest
+            {
+                Line = response.Line,
+                Column = response.Column,
+                FileName = request.FileName,
+                Timeout = request.Timeout,
+                WantMetadata = true
+            });
+        }
+
+        private async Task<GotoDefinitionResponse> GetAliasFromMetadataAsync(GotoDefinitionRequest request)
+        {
+            var document = Workspace.GetDocument(request.FileName);
+            var response = new GotoDefinitionResponse();
+            var lineIndex = request.Line + 3;
+            var column = 0;
+
+            if (document == null)
+            {
+                return response;
+            }
+
+            var semanticModel = await document.GetSemanticModelAsync();
+            var sourceText = await document.GetTextAsync();
+            var sourceLine = sourceText.Lines[lineIndex].ToString();
+            if (sourceLine.Contains("(Context"))
+            {
+                column = sourceLine.IndexOf("(Context", StringComparison.Ordinal);
+            }
+            else
+            {
+                lineIndex = request.Line + 7;
```

Comment:
these arbitrary offsets look scary 😄

Reply:
Yes it is scary, but it has a simple explanation. As GotoDefinition would normally point to the generated DSL, we are instead rerouting to the original extension method (from which the DSL was generated). Ugly, but it works 😄 7 line offset for properties, 3 line offset for methods.